src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -134,13 +134,15 @@ protected function compileColumns(Builder $query, $columns)
             return;
         }
 
-        if ($query->distinct) {
-            $select = 'select distinct ';
-        } else {
-            $select = 'select ';
+        if (! $query->distinct) {
+            return 'select '.$this->columnize($columns);
         }
 
-        return $select.$this->columnize($columns);
+        if(is_bool($query->distinct) or $columns !== ['*']){
```

Comment:
Wouldn't a check for `$query->distinct === true` be more readable at this point?

Reply:
Yes, you are right.
I will fix that in a new commit.